oauth/client_test.go

```diff
@@ -0,0 +1,242 @@
+package oauth
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+	stripe "github.com/stripe/stripe-go"
+	_ "github.com/stripe/stripe-go/testing"
+)
+
+func TestAuthorizeURL(t *testing.T) {
+	stripe.ClientID = "ca_123"
+	url := AuthorizeURL(&stripe.AuthorizeURLParams{})
+
+	assert.Contains(t, url, "https://connect.stripe.com/oauth/authorize?")
+	assert.Contains(t, url, "client_id=ca_123")
+	assert.Contains(t, url, "response_type=code")
+}
+
+func TestAuthorizeURLWithOptionalArgs(t *testing.T) {
+	stripe.ClientID = "ca_123"
+	url := AuthorizeURL(&stripe.AuthorizeURLParams{
+		State:                 stripe.String("test-state"),
+		Scope:                 stripe.String("read_only"),
+		RedirectURI:           stripe.String("https://t.example.com"),
+		ResponseType:          stripe.String("test-code"),
+		StripeLanding:         stripe.String("register"),
+		AlwaysPrompt:          stripe.Bool(true),
+		Express:               stripe.Bool(true),
+		SuggestedCapabilities: stripe.StringSlice([]string{"card_payments"}),
+	})
+
+	assert.Contains(t, url, "https://connect.stripe.com/express/oauth/authorize?")
+	assert.Contains(t, url, "client_id=ca_123")
+	assert.Contains(t, url, "redirect_uri=https%3A%2F%2Ft.example.com")
+	assert.Contains(t, url, "response_type=test-code")
+	assert.Contains(t, url, "scope=read_only")
+	assert.Contains(t, url, "state=test-state")
+	assert.Contains(t, url, "stripe_landing=register")
+	assert.Contains(t, url, "always_prompt=true")
+	assert.Contains(t, url, "suggested_capabilities[0]=card_payments")
+}
+
+func TestAuthorizeURLWithStripeUser(t *testing.T) {
+	stripe.ClientID = "ca_123"
+	var dobDay uint64 = 15
+	var dobMonth uint64 = 10
+	var dobYear uint64 = 2019
+	url := AuthorizeURL(&stripe.AuthorizeURLParams{
+		ResponseType: stripe.String("test-code"),
+		StripeUser: &stripe.OAuthStripeUserParams{
+			BlockKana:          stripe.String("block-kana"),
+			BlockKanji:         stripe.String("block-kanji"),
+			BuildingKana:       stripe.String("building-kana"),
+			BuildingKanji:      stripe.String("building-kanji"),
+			BusinessName:       stripe.String("b-name"),
+			BusinessType:       stripe.String("llc"),
+			City:               stripe.String("Elko"),
+			Country:            stripe.String("US"),
+			Currency:           stripe.String("USD"),
+			DOBDay:             &dobDay,
+			DOBMonth:           &dobMonth,
+			DOBYear:            &dobYear,
```

Comment:
And switching those to `int64` will also allow you to use the `string.Int64` helper down here.

Reply:
Sweet. I was wondering why that type didn't have a helper.